keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,390 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.ServiceClient;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.rest.RestResponse;
+import com.azure.common.http.rest.RestVoidResponse;
+import com.azure.common.http.rest.SimpleRestResponse;
+import com.azure.common.implementation.RestProxy;
+import com.azure.keyvault.implementation.Page;
+import com.azure.keyvault.implementation.RestPagedResponseImpl;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretAttributes;
+import org.apache.commons.lang3.Validate;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class SecretClient extends ServiceClient {
+    static final String API_VERSION = "7.0";
+    static final String ACCEPT_LANGUAGE = "en-US";
+    static final int DEFAULT_MAX_PAGE_RESULTS = 25;
+    static final String CONTENT_TYPE_HEADER_VALUE = "application/json";
+
+
+    private String vaultEndPoint;
+    private final SecretService service;
+
+    /**
+     * Creates a SecretClient that uses {@code pipeline} to
+     * service requests
+     *
+     * @param vaultEndPoint URL for the Azure KeyVault service.
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    SecretClient(URL vaultEndPoint, HttpPipeline pipeline) {
```

Comment:
In general, update `EndPoint` to `Endpoint`.

Reply:
updated.